src/main/java/fi/aalto/cs/intellij/common/Course.java

```diff
@@ -0,0 +1,219 @@
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+public class Course {
+  @NotNull
+  private final String name;
+
+  @NotNull
+  private final List<Module> modules;
+
+  // Maps ids of required plugins to the names of required plugins.
+  @NotNull
+  private final Map<String, String> requiredPlugins;
+
+  /**
+   * Constructs a course with the given parameters. Course objects are usually created using
+   * {@link Course#fromConfigurationFile}
+   *
+   * @param name            The name of the course
+   * @param modules         The list of modules in the course.
+   * @param requiredPlugins A map containing the required plugins for this course. The keys are the
+   *                        ids of the plugins and the values are the names of the plugins.
+   */
+  public Course(@NotNull String name, @NotNull List<Module> modules,
+                @NotNull Map<String, String> requiredPlugins) {
+    this.name = name;
+    this.modules = modules;
+    this.requiredPlugins = requiredPlugins;
+  }
+
+  /**
+   * Returns an "empty" course, that is, a course with no name, no modules, and no required plugins.
+   * @return An empty course.
+   */
+  public static Course createEmptyCourse() {
+    return new Course("", new ArrayList<>(), new HashMap<>());
+  }
+
+  /**
+   * Parses the course configuration file located at the given path.
+   *
+   * @param pathToCourseConfig The path to the course configuration file.
+   * @return A course instance containing the information parsed from the configuration file.
+   * @throws FileNotFoundException                     If the configuration file can't be found with
+   *                                                   the given path.
+   * @throws MalformedCourseConfigurationFileException If the configuration file is malformed in
+   *                                                   any way.
+   */
+  public static Course fromConfigurationFile(@NotNull String pathToCourseConfig)
+      throws FileNotFoundException, MalformedCourseConfigurationFileException {
+    JSONObject jsonObject = getCourseJsonObject(pathToCourseConfig);
+    String courseName = getCourseName(jsonObject, pathToCourseConfig);
+    List<Module> courseModules = getCourseModules(jsonObject, pathToCourseConfig);
+    Map<String, String> requiredPlugins = getCourseRequiredPlugins(jsonObject, pathToCourseConfig);
+    return new Course(courseName, courseModules, requiredPlugins);
+  }
+
+
+  /**
+   * Returns the name of the course.
+   *
+   * @return The name of the course.
+   */
+  @NotNull
+  String getName() {
+    return name;
+  }
+
+  /**
+   * Returns the names of all the modules in the course. If the course object is created with
+   * {@link Course#fromConfigurationFile}, then the module names are returned in the order
+   * in which they are listed in the course configuration file.
+   */
+  @NotNull
+  public List<String> getModuleNames() {
```

Comment:
Is this method more about presentation logic (and thus its place is perhaps not in a model class)? Could this class have a getter for a list of module objects instead? But this is a very minor issue (if an issue at all).

Reply:
Good point! I'll make this  `List<Module> getModules` and move the name-mapping logic to the UI code